File: requests.cc
#include "requests.pb.cc"

#include <iostream>

using namespace std;

int main() {
  req::Request r;

  // Add the tagging extension.
  // r.MutableExtension(req::TaggingRequest::tagging);
  // both work because Clear() affects the content of the extension and not
  // that of `r'
  r.MutableExtension(req::TaggingRequest::tagging)->Clear();
  cout << "has tagging extension " << r.HasExtension(req::TaggingRequest::tagging) << endl;
  
  // Add the numbered extension and set its `number' field.
  req::NumberedRequest* numbered = r.MutableExtension(req::NumberedRequest::numbered);
  cout << "has numbered extension "
       << r.HasExtension(req::NumberedRequest::numbered) << endl;
  numbered->set_number(42);
  cout << "NumberedRequest field "
       << ((r.GetExtension(req::NumberedRequest::numbered)).number()) << endl;
}

File: requests.py
from requests_pb2 import *

def demo():
  r = Request()
  # Try to add the tagging extension.
  # r.Extensions[TaggingRequest.tagging]
  r.Extensions[TaggingRequest.tagging].Clear()
  try:
    r.Extensions[TaggingRequest.tagging].SetInParent()
  except AttributeError as e:
    print e  # 'TaggingRequest' object has no attribute 'SetInParent'

  print 'has tagging extension', r.HasExtension(TaggingRequest.tagging)

  # Add the numbered extension by setting its `number' field.
  r.Extensions[NumberedRequest.numbered].number = 42
  print 'has numbered extension', r.HasExtension(NumberedRequest.numbered)
  print 'NumberedRequest field', r.Extensions[NumberedRequest.numbered].number

demo()
